packages/toolkit/src/query/tests/buildHooks.test.tsx

```diff
@@ -1977,47 +1976,49 @@ describe('hooks with createApi defaults set', () => {
       posts = [...initialPosts]
 
       const handlers = [
-        rest.get('https://example.com/posts', (req, res, ctx) => {
-          return res(ctx.json(posts))
+        http.get('https://example.com/posts', () => {
+            return HttpResponse.json(posts)
         }),
-        rest.put<Partial<Post>>(
+        http.put<Post, Partial<Post>>(
           'https://example.com/post/:id',
-          (req, res, ctx) => {
-            const id = Number(req.params.id)
+          async ({ request, params }) => {
+              const body = await request.json();
+            const id = Number(params.id)
             const idx = posts.findIndex((post) => post.id === id)
 
             const newPosts = posts.map((post, index) =>
               index !== idx
                 ? post
                 : {
-                    ...req.body,
+                    ...body,
                     id,
-                    name: req.body.name || post.name,
+                    name: body?.name || post.name,
                     fetched_at: new Date().toUTCString(),
                   }
             )
             posts = [...newPosts]
 
-            return res(ctx.json(posts))
+              return HttpResponse.json(posts)
           }
         ),
-        rest.post('https://example.com/post', (req, res, ctx) => {
-          let post = req.body as Omit<Post, 'id'>
+        http.post<any, Omit<Post, 'id'>>('https://example.com/post', async ({ request }) => {
+            const body = await request.json();
+          let post = body
           startingId += 1
           posts.concat({
             ...post,
             fetched_at: new Date().toISOString(),
             id: startingId,
           })
-          return res(ctx.json(posts))
+            return HttpResponse.json(posts)
         }),
       ]
 
       server.use(...handlers)
     })
 
     interface Post {
-      id: number
+      id: string
```

Comment:
why make this change if it forces you to add so many `as any`s? The server response id is a number, so I don't understand why it's changed here.

Reply:
@EskiMojo14  The reason why I changed the type for `Post` to `string` is because MSW 2 has the `PathParams` type like this:

```ts
type PathParams<KeyType extends keyof any = string> = {
    [ParamName in KeyType]: string | ReadonlyArray<string>;
};
```

so if you have something like:

```ts
{ id: number }
```

it'll yell at you, and I didn't wanna mess with runtime unit tests too much just to make TS happy.


![Screenshot 2024-01-23 125845](https://github.com/reduxjs/redux-toolkit/assets/86934138/84a6dab7-59f9-4e78-b085-7d3fa8513061)